bugbug/tools/code_review.py

```diff
@@ -0,0 +1,281 @@
+
+PROMPT_TEMPLATE_FILTERING_ANALYSIS = """Please, double check the code review provided for the patch below.
+Just report the comments that are:
+- applicable for the patch;
+- consistent with the source code in Firefox;
+- focusing on reporting possible bugs, major readability regressions, or similar concerns.
+
+Do not change the contents of the comments and the report format.
+Adopt the template below:
+[
+    {{
+        "file": "com/br/main/Pressure.java",
+        "code_line": 458,
+        "comment" : "In the third code block, you are using `nsAutoStringN<256>` instead of `nsString`. This is a good change as `nsAutoStringN<256>` is more efficient for small strings. However, you should ensure that the size of `tempString` does not exceed 256 characters, as `nsAutoStringN<256>` has a fixed size."
+    }}
+]
+
+Review:
+{review}
+
+Patch:
+{patch}
+
+As examples of not expected comments, not related to the current patch, please, check some below:
+    - Please note that these are minor improvements and the overall quality of the patch is good. The documentation is being expanded in a clear and structured way, which will likely be beneficial for future development.
+    - Please note that these are just suggestions and the code might work perfectly fine as it is. It's always a good idea to test all changes thoroughly to ensure they work as expected.
+    - Overall, the patch seems to be well implemented with no major concerns. The developers have made a conscious decision to align with Chrome's behavior, and the reasoning is well documented.
+    - There are no complex code changes in this patch, so there's no potential for major readability regressions or bugs introduced by the changes.
+    - The `focus(...)` method is called without checking if the element and its associated parameters exist or not. It would be better to check if the element exists before calling the `focus()` method to avoid potential errors.
+    - It's not clear if the `SearchService.sys.mjs` file exists or not. If it doesn't exist, this could cause an error. Please ensure that the file path is correct."""
+
+
+class ReviewRequest:
+    patch_id: int
+
+    def __init__(self, patch_id) -> None:
+        super().__init__()
+        self.patch_id = patch_id
+
+
+class Patch:
+    raw_diff: str
+
+    def __init__(self, raw_diff) -> None:
+        super().__init__()
+        self.raw_diff = raw_diff
+
+
+class ReviewData(ABC):
+    @abstractmethod
+    def get_review_request_by_id(self, review_id: int) -> ReviewRequest:
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_patch_by_id(self, patch_id: int) -> Patch:
+        raise NotImplementedError
+
+
+class PhabricatorReviewData(ReviewData):
+    def __init__(self):
+        phabricator.set_api_key(
+            get_secret("PHABRICATOR_URL"), get_secret("PHABRICATOR_TOKEN")
+        )
```

Comment:
It would be great if we could avoid global status, but this is out of the scope of this PR. We could take care of it later.